Feature/accept serialized binary array (#17)
scalar-labs/scalardl-node-client-sdk: README.md

};
```

## Create raw gRPC requests

> Users can use SDK to create raw gRPC requests in byte arries (JavaScript Uint8Array) too.

Reviewer: ```suggestion
You can also create a raw gRPC request in byte array (JavaScript Uint8Array) too.
```

> The functions have different names but the parameters are exactly identical.

Reviewer: ```suggestion
Note that the name of functions are different from usual functions such as `executeContract` but the parameters are exactly the same.
```

### Register the certificate
```javascript
const binary = await ClientService.createSerializedCertificateRegistrationRequest();
```

### Register contracts
```javascript
const binary = await clientService.createSerializedContractRegistrationRequest('contractId', 'com.example.contract.contractName', contractUint8Array, propertiesObject);
```

### Register functions
```javascript
const binary = await clientService.createSerializedFunctionRegistrationRequest('functionId', 'com.example.function.functionName', functionUint8Array);
```

### List registered contracts
```javascript
const binary = await clientService.createSerializedContractsListingRequest();
```

### Execute a contract
```javascript
const binary = await clientService.createSerializedContractExecutionRequest('contractId', argumentObject);
```

### Validate an asset
```javascript
const binary = await clientService.createSerializedLedgerValidationRequest('assetId');
```

## Send the raw gRPC requests to Scalar DL servers

> The SDK provides another `ClientServiceWithBinary` class whereby users can send byte arries of requests to Scalar DL servers.

Reviewer: ```suggestion
The SDK has another `ClientServiceWithBinary` class for you to send the byte array of a request to Scalar DL network.
```

It is similar with `ClientService` class to construct `ClientServiceWithBinary` class.
However, ClientServiceWithBinary is different from ClientService.

> ClientService encapsulates responses into Scalar defined objects but ClientServiceWithBinary returns gRPC native objects.

Reviewer: Oh ,I think they should return the same object even if serialized byte array is an input.

### Register the certificate
```javascript
const binary = await ClientService.createSerializedCertificateRegistrationRequest();
```

### Register contracts
```javascript
const binary = await clientService.createSerializedContractRegistrationRequest('contractId', 'com.example.contract.contractName', contractUint8Array, propertiesObject);
```

### Register functions
```javascript
const binary = await clientService.createSerializedFunctionRegistrationRequest('functionId', 'com.example.function.functionName', functionUint8Array);
```

### List registered contracts
```javascript
const binary = await clientService.createSerializedContractsListingRequest();
```

### Execute a contract
```javascript
const binary = await clientService.createSerializedContractExecutionRequest('contractId', argumentObject);
```

### Validate an asset
```javascript
const binary = await clientService.createSerializedLedgerValidationRequest('assetId');
```

## Send the raw gRPC requests to Scalar DL servers
The SDK has another `ClientServiceWithBinary` class for you to send the byte array of a request to Scalar DL network.

> It is similar with `ClientService` class to construct `ClientServiceWithBinary` class.

Reviewer: I think we can delete this line without losing ease of understanding. :)

Reviewer 2: @feeblefakie 
Revised.
By the way, could you publish SDK base version 2.4.5 (which fixed the issue of optional contract arguments)?
Then, I can update it in the PR as well :)
Thank you.

Reviewer: Sorry, it is just published !

Reviewer 2: > Sorry, it is just published !

@feeblefakie 
Thank you.
Updated.
Please review again :)
Sorry for taking so long.
